cmd/deploy/deploy.go

```diff
@@ -0,0 +1,458 @@
+package deploy
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	dtypes "github.com/docker/docker/api/types"
+	dclient "github.com/docker/docker/client"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/kind/pkg/cluster"
+	klog "sigs.k8s.io/kind/pkg/log"
+)
+
+func New() *cobra.Command {
+	deployCmd := &cobra.Command{
+		Use:   "deploy <deployment yaml>",
+		Short: "Deploy cluster.",
+		RunE:  deployFn,
+	}
+	return deployCmd
+}
+
+type ClusterSpec struct {
+	Kind string    `yaml:"kind"`
+	Spec yaml.Node `yaml:"spec"`
+}
+
+type IngressSpec struct {
+	Kind string    `yaml:"kind"`
+	Spec yaml.Node `yaml:"spec"`
```

Comment:
What is the idea behind using yaml.Node and not kindSpec for this field?

Reply:
i leave this to marshalled based on the kind of the spec
so while this currently supports 
kind cluster + metallb + meshnet

you could write a config and spec implementation for say
Local cluster + calico as ingress + multis 
and it would stay the same form of config